lib/JSON/Schema.pm6

```diff
@@ -375,9 +375,13 @@ class JSON::Schema {
     my class RequiredCheck does Check {
         has Str @.prop;
         method check($value --> Nil) {
-            if $value ~~ Associative && not [&&] $value{@!prop}.map(*.defined) {
-                die X::JSON::Schema::Failed.new:
-                    :$!path, :reason("Object does not have required property");
+            if $value ~~ Associative {
+                for @.prop -> $mandatory_prop {
+                    unless $value{$mandatory_prop}:exists {
+                        die X::JSON::Schema::Failed.new:
+                            :$!path, :reason("Missing required property '$mandatory_prop'");
+                    }
+                }
```

Comment:
Otherwise I do not see if as not elegant in any way. It is, in fact, much better than it was, because for one-liner I have sacrificed some clarity in this place. Your code is much more readable for me.

Reply:
I think I'd probably write it as `if @!prop.grep({ not $value{$_}.defined }) -> @missing { ... }` and then make the reason be `"Missing required properties: @missing.map({ qq/'$_'/ }).join(', ')"`, to provide all of the names up front.